src/api/Interpret.cc

```diff
@@ -217,29 +217,20 @@ void Interpret::interp(ASTNode& n) {
             }
             case t_assert: {
                 if (isInitialized()) {
-                    sstat status;
-                    ASTNode &asrt = **(n.children->begin());
+                    ASTNode const & asrt = **(n.children->begin());
                     LetRecords letRecords;
                     PTRef tr = parseTerm(asrt, letRecords);
                     if (tr == PTRef_Undef)
                         notify_formatted(true, "assertion returns an unknown sort");
                     else {
                         assertions.push(tr);
-                        char *err_msg = NULL;
-                        status = main_solver->insertFormula(tr, &err_msg);
-
-                        if (status == s_Error)
-                            notify_formatted(true, "Error");
-                        else if (status == s_Undef)
-                            notify_success();
-                        else if (status == s_False)
+                        if (logic->getSortRef(tr) != logic->getSort_bool()) {
+                            auto errorMessage = "Top-level assertion sort must be Bool, got " + logic->printSort(logic->getSortRef(tr));
+                            notify_formatted(true, errorMessage.c_str());
+                        } else {
+                            main_solver->insertFormula(tr);
```

Comment:
Maybe
```suggestion
                        try {
                            main_solver->insertFormula(tr);
                            notify_success();
                        } catch (OsmtApiException const & e) {
                            notify_formatted(true, e.msg.c_str());
                        }

```
(modulo my c++ syntax & other errors...)

Reply:
Fixed and added a test for this as well, just in case.